### DISCUSSION TITLE
allowAudioPlaybackCapture & apk version

### DISCUSSION THREAD

**Discussion by JuanCaiza** [airsdk/Adobe-Runtime-Support#2996] [Q&A] [answered]
Hello everyone,
I have 2 questions.
1- I create automatic apk files in the form of call adt -package -target, but do I have the chance to change it directly through the code instead of opening the xml file and changing the version of the application every time?

2- Is there any chance to change allowAudioPlaybackCapture=true on the application?
Is there any chance to add as3 code that can be in other options?
https://developer.android.com/media/platform/av-capture#capture_policy

Thanks

**Comment by JuanCaiza** [comment]
@ajwfrost any tips??

**Reply by ajwfrost** [reply, marked as answer]
Hi

Sorry, I missed the earlier notification on this thread..

1) The version code in the APK file is just taken from the application descriptor XML `versionNumber` field.. if you need to change it, you'd need to have some preprocessor on this perhaps?

We do this for AIR SDK Manager, the version is in a config file and we use Ant to build, with something like the below to pre-process the app descriptor:
```
                <copy file="${app.descriptor.template}" tofile="${app.descriptor}" overwrite="true">
                        <filterchain>
                                <replacestring from="0.0.0" to="${version}" />
                        </filterchain>
                </copy>
```

2) From that link, it sounds like the audio capture policy can be adjusted from the Android manifest? per your comment about `allowAudioPlaybackCapture=true` .. so could you add this to your manifestAdditions section of the app descriptor file?
Looks like this should be within the `application` definition, and you would need to ensure you're using build tools 28 or higher which I would hope is the case..

For an example, see https://airsdk.dev/docs/building/application-descriptor-files/elements/android#manifestadditions

thanks

**Reply by JuanCaiza** [reply]
Thank you very much. You are amazing.
